metrics-server/src/chain.rs

```diff
@@ -0,0 +1,96 @@
+use prometheus_client::registry::Registry;
+
+use parking_lot::RwLock;
+
+use crate::NumericClosureMetric;
+use nimiq_blockchain::{AbstractBlockchain, Blockchain};
+use std::sync::Arc;
+
+pub struct BlockMetrics {}
+
+impl BlockMetrics {
+    pub fn register(registry: &mut Registry, blockchain: Arc<RwLock<Blockchain>>) {
+        BlockMetrics::register_staking(registry, blockchain.clone());
+        BlockMetrics::register_accounts_tree(registry, blockchain.clone());
+        BlockMetrics::register_chain(registry, blockchain.clone());
+        BlockMetrics::register_block_stats(registry, blockchain);
+    }
+
+    fn register_staking(registry: &mut Registry, blockchain: Arc<RwLock<Blockchain>>) {
+        let sub_registry = registry.sub_registry_with_prefix("staking");
+
+        let bc = blockchain.clone();
+        let closure = Box::new(NumericClosureMetric::new_gauge(Box::new(move || {
+            bc.read().get_staking_contract().active_validators.len() as u32
+        })));
+        sub_registry.register("active_validators", "Number of active validators", closure);
+
+        let closure = Box::new(NumericClosureMetric::new_gauge(Box::new(move || {
+            blockchain.read().get_staking_contract().parked_set.len() as u32
+        })));
+        sub_registry.register("parked_validators", "Number of parked validators", closure);
+    }
+
+    fn register_accounts_tree(registry: &mut Registry, blockchain: Arc<RwLock<Blockchain>>) {
+        let sub_registry = registry.sub_registry_with_prefix("accounts");
+
+        let closure = Box::new(NumericClosureMetric::new_gauge(Box::new(move || {
+            blockchain.read().state.accounts.size(None) as u32
+        })));
+        sub_registry.register("size", "Size of accounts tree", closure);
```

Comment:
I'd call this metric `accounts_tree_size` instead to avoid any other confusion when using this metric

Reply:
Ahh, `nimiq_accounts_size` sounds good. Nvm then